### PR TITLE
[Internal] Encryption project: Adds gates to prevent breaking changes

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

This adds a gate to prevent SDK changes that break encryption project. This way it is always safe to do an encryption release.

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: templates/build-preview.yml [hunk 1]
```diff
@@ -27,6 +27,25 @@ jobs:
...
+    clean: true  # if true, execute `execute git clean -ffdx && git reset --hard HEAD` before fetching
+
+  - task: DotNetCoreCLI@2
+    displayName: Build Microsoft.Azure.Cosmos.Encryption Project Ref
+    inputs: 
+      command: build  
+      configuration: $(parameters.BuildConfiguration)
+      nugetConfigPath: NuGet.config
+      projects: Microsoft.Azure.Cosmos.sln 
+      arguments: -p:Optimize=true -p:IsPreview=true;SdkProjectRef=true
```
Comment (ealsur) [inline templates/build-preview.yml:46]: Neat!

File: templates/build-preview.yml [hunk 2]
```diff
@@ -27,6 +27,25 @@ jobs:
...
+  - checkout: self  # self represents the repo where the initial Pipelines YAML file was found
+    clean: true  # if true, execute `execute git clean -ffdx && git reset --hard HEAD` before fetching
+
+  - task: DotNetCoreCLI@2
+    displayName: Build Microsoft.Azure.Cosmos.Encryption Project Ref
+    inputs: 
+      command: build  
+      configuration: $(parameters.BuildConfiguration)
+      nugetConfigPath: NuGet.config
+      projects: Microsoft.Azure.Cosmos.sln
```
Comment (ealsur) [inline templates/build-preview.yml:45]: Would it make sense just to build the Encryption CSPROJ?

**Review by ealsur — APPROVED** [review]

File: templates/build-preview.yml (hunk 2 above)
Comment (FabianMeiswinkel) [reply templates/build-preview.yml:45]: I would keep it scoped to solution in case we want similar checks going forward for other projects.

**Review by FabianMeiswinkel — DISMISSED** [review]
Very good idea!

File: templates/build-preview.yml (hunk 2 above)
Comment (j82w) [reply templates/build-preview.yml:45]: I agree with keeping at the solution so tests also are built.

**Review by abhijitpai — APPROVED** [review]

**Review by FabianMeiswinkel — APPROVED** [review]
Thanks!!!

**Pull request merged**